pkg/vpcmodel/explainabilityPrint.go

```diff
@@ -58,10 +58,29 @@ func connHeader(connQuery *netset.TransportSet) string {
 	return ""
 }
 
+// nodes are externals
+func getExternalTypeStr(userInput string, nodes []Node) string {
+	externalType := ""
+	for _, node := range nodes {
+		if node.IsPublicInternet() {
+			if externalType != "" && externalType != publicInternetNodeName {
+				return fmt.Sprintf("%s (%s, %s)", userInput, externalType, publicInternetNodeName)
+			}
+			externalType = publicInternetNodeName
+		} else {
+			if externalType != "" && externalType != serviceNetworkNodeName {
+				return fmt.Sprintf("%s (%s, %s)", userInput, serviceNetworkNodeName, externalType)
+			}
+			externalType = serviceNetworkNodeName
+		}
+	}
+	return fmt.Sprintf("%s (%s)", userInput, externalType)
+}
```

Comment:
```suggestion
func getExternalTypeStr(userInput string, nodes []Node) string {
	var hasPublicNode, hasServiceNode bool
	var externalTypes []string
	for _, node := range nodes {
		if node.IsPublicInternet() {
			if !hasPublicNode {
				hasPublicNode = true
				externalTypes = append(externalTypes, publicInternetNodeName)
			}
		} else {
			if !hasServiceNode {
				hasServiceNode = true
				externalTypes = append(externalTypes, serviceNetworkNodeName)
			}
		}
	}
	return fmt.Sprintf("%s (%s)", userInput, strings.Join(externalTypes, comma))
}
```
A somewhat cleaner solution